src/Strings.ts

```diff
@@ -307,15 +322,13 @@ export const strings = (locale: string) => {
         locale = 'el-gr';
     else if (locale.startsWith('it'))
         locale = 'it-it';
-    else if (locale === 'zh-yue')
-        locale = 'zh-yue';
```

Comment:
I found same case. So, I removed it.

Reply:
Is there any reason why `zh-yue` is removed?

Reply:
My fault for approving this. I was complaining about code of this form:

```ts 
if (locale === 'foo')
    locale = 'foo';
```

because I found it confusing that the action was to assign locale to its current value. But obviously its important to catch this case.

I think a better version of this part of the logic is:

```ts
    ...
    else if (locale === 'zh-hant' || locale === 'zh-yue') {
        // leave alone
    } else if (locale === 'zh-hk' || locale === 'zh-mo' || locale === 'zh-tw')
        locale = 'zh-hant';
    else if (locale.startsWith('zh'))
        locale = 'zh-hans';
    ...
```

That way it's obvious what's going on.

However I think an even better fix is to break out the locale mapping logic into its own function so that the mapping is handled by a return rather than an assignment, which is more functional and (to my eye) clearer:

```ts
export const mapLocale = (locale: string) => {
    if (locale.startsWith('da'))
        return 'da-dk';
    if (locale.startsWith('de'))
        return 'de-de';
    ...
    if locale.startsWith('zh')) {
        if (locale === 'zh-hant' || locale === 'zh-yue')
            return locale;
        if (locale === 'zh-hk' || locale === 'zh-mo' || locale === 'zh-tw')
            return 'zh-hant';
        return 'zh-hans';
    }
    ...
    if (locale in localizedStrings)
        return locale;
    return 'en-us';
}

export const strings = (locale: string) => localizedStrings[mapLocale(locale)];
```

Reply:
💗 to make it functional programming. Let me make a PR for this.

Reply:
#804 is the PR which fix this, along with your proposed `mapLocale` function.

Reply:
@compulim @billba I missed the conversation.. Thank you for fixing the fault. It has changed to be more readable 😃